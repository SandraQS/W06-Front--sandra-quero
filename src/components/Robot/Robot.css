.card-robot {
  padding: 20px;
}
.card-robot_container-image {
  width: 100%;
  height: 58vh;
}
.card-robot__image {
  width: 100%;
  height: 58vh;
  object-fit: cover;
}

.card-robot__title {
  font-size: 35px;
}

.card-robot__title-features {
  font-size: 25px;
  color: gray;
  font-weight: normal;
  padding-left: 20px;
}

.card-robot__list-features {
  list-style: none;
}
.card-robot__feature {
  border-left: 2px brown solid;
  padding: 3px;
  padding-left: 10px;
  margin-bottom: 10px;
}
