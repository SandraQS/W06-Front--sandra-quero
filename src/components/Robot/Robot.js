const Robot = ({ infoRobot }) => {
  return (
    <article className="card-robot">
      <img
        src={infoRobot.imagen}
        alt={infoRobot.nombre}
        className="card-robot__image"
        whith="150"
      />
      <h2 className=" card-robot__title">{infoRobot.nombre}</h2>
      <h3 className="card-robot__title-features">
        Características de {infoRobot.nombre}
      </h3>
      <ul className="card-robot__list-features">
        <li className="card-robot__feature">
          Velocidad: {infoRobot.caracteristicas.velocidad}
        </li>
        <li className="card-robot__feature">
          Resistencia: {infoRobot.caracteristicas.resistencia}
        </li>
        <li className="card-robot__feature">
          Fecha de creación: {infoRobot.caracteristicas.creacion}
        </li>
      </ul>
      <button type="button" className="btn btn-dark card-robot__edit">
        Editar
      </button>
      <button type="button" className="btn btn-danger card-robot__delete">
        Eliminar
      </button>
    </article>
  );
};

export default Robot;
