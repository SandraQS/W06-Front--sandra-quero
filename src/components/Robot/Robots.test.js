import { getByRole, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import ReactTestRenderer from "react-test-renderer";
import Robot from "./Robot";

describe("Given Robots component", () => {
  describe("When it's rendered", () => {
    test("Then it should match with de last snapshot", () => {
      const robot = {
        caracteristicas: {
          velocidad: 5,
          resistencia: 9,
          creacion: "1996-05-19T22:00:00.000Z",
        },
        _id: "61855ad154ce63991d588ddb",
        nombre: "Aire",
        imagen:
          "https://cdn.hobbyconsolas.com/sites/navi.axelspringer.es/public/styles/hc_1440x810/public/media/image/2020/04/astro-robot-ps5-1920415.jpg?h=d1cb525d&itok=NpaKRK-Q",
      };

      const robotCard = ReactTestRenderer.create(
        <Robot infoRobot={robot} key={robot._id} />
      );

      expect(robotCard.toJSON()).toMatchSnapshot();
    });
  });
  describe("When click in button delete", () => {
    test("Then it should call clickDelete function", async () => {
      const clickDelete = jest.fn();
      screen.debug();
      const buttonDelete = await screen.findByRole("button");

      // buttonDelete.addEventListener("click", clickDelete);
      userEvent.click(buttonDelete);
      expect(clickDelete).toBeCalled();
    });
  });
});
