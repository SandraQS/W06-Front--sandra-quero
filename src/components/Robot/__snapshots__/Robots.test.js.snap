// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given Robots component When it's rendered Then it should match with de last snapshot 1`] = `
<article
  className="card-robot"
>
  <img
    alt="Aire"
    className="card-robot__image"
    src="https://cdn.hobbyconsolas.com/sites/navi.axelspringer.es/public/styles/hc_1440x810/public/media/image/2020/04/astro-robot-ps5-1920415.jpg?h=d1cb525d&itok=NpaKRK-Q"
    whith="150"
  />
  <h2
    className=" card-robot__title"
  >
    Aire
  </h2>
  <h3
    className="card-robot__title-features"
  >
    Características de 
    Aire
  </h3>
  <ul
    className="card-robot__list-features"
  >
    <li
      className="card-robot__feature"
    >
      Velocidad: 
      5
    </li>
    <li
      className="card-robot__feature"
    >
      Resistencia: 
      9
    </li>
    <li
      className="card-robot__feature"
    >
      Fecha de creación: 
      1996-05-19T22:00:00.000Z
    </li>
  </ul>
  <button
    className="btn btn-dark card-robot__edit"
    type="button"
  >
    Editar
  </button>
  <button
    className="btn btn-danger card-robot__delete"
    type="button"
  >
    Eliminar
  </button>
</article>
`;
